solr/core/src/test/org/apache/solr/response/TestPrometheusResponseWriterCloud.java

```diff
@@ -0,0 +1,170 @@
+import org.apache.solr.client.solrj.SolrRequest.METHOD;
+import org.apache.solr.client.solrj.SolrRequest.SolrRequestType;
+import org.apache.solr.client.solrj.impl.InputStreamResponseParser;
+import org.apache.solr.client.solrj.request.CollectionAdminRequest;
+import org.apache.solr.client.solrj.request.GenericSolrRequest;
+import org.apache.solr.cloud.SolrCloudTestCase;
+import org.apache.solr.common.params.ModifiableSolrParams;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.util.SolrJettyTestRule;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class TestPrometheusResponseWriterCloud extends SolrCloudTestCase {
+  @ClassRule public static SolrJettyTestRule solrClientTestRule = new SolrJettyTestRule();
+
+  @BeforeClass
+  public static void setupCluster() throws Exception {
+    System.setProperty("metricsEnabled", "true");
+    configureCluster(1)
+        .addConfig(
+            "config", TEST_PATH().resolve("configsets").resolve("cloud-minimal").resolve("conf"))
+        .configure();
+  }
+
+  @Before
+  public void ensureCollectionsExist() throws Exception {
+    SolrClient client = cluster.getSolrClient();
+    try {
+      CollectionAdminRequest.deleteCollection("collection1").process(client);
+    } catch (Exception ignored) {
+    }
+    try {
+      CollectionAdminRequest.deleteCollection("collection2").process(client);
+    } catch (Exception ignored) {
+    }
+
+    CollectionAdminRequest.createCollection("collection1", "config", 1, 1).process(client);
+    CollectionAdminRequest.createCollection("collection2", "config", 1, 1).process(client);
+    cluster.waitForActiveCollection("collection1", 1, 1);
+    cluster.waitForActiveCollection("collection2", 1, 1);
+  }
+
+  @Test
+  public void testPrometheusCloudLabels() throws Exception {
+    var solrClient = cluster.getSolrClient();
+
+    // Increment solr_metrics_core_requests metric for /select
+    SolrQuery query = new SolrQuery("*:*");
+    solrClient.query("collection1", query);
+
+    var req =
+        new GenericSolrRequest(
+            METHOD.GET,
+            "/admin/metrics",
+            SolrRequestType.ADMIN,
+            new ModifiableSolrParams().set("wt", "prometheus"));
+    req.setResponseParser(new InputStreamResponseParser("prometheus"));
+
+    NamedList<Object> prometheusResponse = solrClient.request(req);
+    assertNotNull("null response from server", prometheusResponse);
+    NamedList<Object> res = solrClient.request(req);
+    InputStream in = (InputStream) prometheusResponse.get("stream");
+
+    assertNotNull("null response from server", res);
+    String output = new String(in.readAllBytes(), StandardCharsets.UTF_8);
+
+    output
+        .lines()
+        .forEach(
+            line -> {
+              if (line.startsWith("solr_metrics_core_requests_total")
+                  && line.contains("handler=\"/select\""))
+                assertTrue(
```

Comment:
we should check that at least one line reaches this point :-)
Maybe use the `anyMatch` technique that you use later.